chain/rpc/src/rpc.rs

```diff
@@ -685,4 +771,85 @@ mod tests {
+        let anvil = chain_types::utils::create_anvil(Some(expected_block_time));
+        let chain_key_0 = ChainKeypair::from_secret(anvil.keys()[0].to_bytes().as_ref())?;
+        let node_address: H160 = chain_key_0.public().to_address().into();
+
+        // Deploy contracts
+        let (contract_instances, module, safe) = {
+            let client = create_rpc_client_to_anvil(SurfRequestor::default(), &anvil, &chain_key_0);
+            let instances = ContractInstances::deploy_for_testing(client.clone(), &chain_key_0).await?;
+
+            // deploy MULTICALL contract to anvil
+            deploy_multicall3_to_anvil(client.clone()).await?;
+
+            let (module, safe) = chain_types::utils::deploy_one_safe_one_module_and_setup_for_testing(
+                &instances,
+                client.clone(),
+                &chain_key_0,
+            )
+            .await?;
+
+            // deploy a module and safe instance and add node into the module. The module is enabled by default in the safe
+            (instances, module, safe)
+        };
+
+        let cfg = RpcOperationsConfig {
+            chain_id: anvil.chain_id(),
+            tx_polling_interval: Duration::from_millis(10),
+            expected_block_time,
+            finality: 2,
+            contract_addrs: ContractAddresses::from(&contract_instances),
+            module_address: module,
+            safe_address: safe,
+            ..RpcOperationsConfig::default()
+        };
+
+        let client = JsonRpcProviderClient::new(
+            &anvil.endpoint(),
+            SurfRequestor::default(),
+            SimpleJsonRpcRetryPolicy::default(),
+        );
+
+        // Wait until contracts deployments are final
+        sleep((1 + cfg.finality) * expected_block_time).await;
+
+        let rpc = RpcOperations::new(client.clone(), &chain_key_0, cfg)?;
+
+        // check the eligibility status (before registering in the NetworkRegistry contract)
+        let result_before_register_in_the_network_registry = rpc.get_eligibility_status(node_address.into()).await?;
+
+        assert!(
+            !result_before_register_in_the_network_registry,
+            "node should not be eligible"
+        );
+
+        // register the node
+        match &rpc.contract_instances.network_registry_proxy {
+            NetworkRegistryProxy::Dummy(e) => {
+                let _ = e.owner_add_account(cfg.safe_address.into()).send().await?.await?;
+            }
+            NetworkRegistryProxy::Safe(e) => {}
+        };
+
+        let _ = rpc
+            .contract_instances
+            .network_registry
+            .manager_register(vec![cfg.safe_address.into()], vec![node_address])
+            .send()
+            .await?
+            .await?;
+
+        // check the eligibility status (after registering in the NetworkRegistry contract)
+        let result_after_register_in_the_network_registry = rpc.get_eligibility_status(node_address.into()).await?;
```

Comment:
There are 3 parts of the multicall, but the eligibility status is checked only after changing 2 conditions.

Reply:
No state gets changed in the call. It first check eligibility. If revert, uses other returned results to compute eligibility